x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/classification_exploration/column_data.tsx

```diff
@@ -4,79 +4,80 @@
-      (pc: PredictedClass) => pc.predicted_class === classData.actual_class
-    );
-    const incorrectlyPredictedClass = classData.predicted_classes.find(
-      (pc: PredictedClass) => pc.predicted_class !== classData.actual_class
-    );
-
-    let accuracy;
-    if (correctlyPredictedClass !== undefined) {
-      accuracy = correctlyPredictedClass.count / classData.actual_class_doc_count;
-      // round to 2 decimal places without converting to string;
-      accuracy = Math.round(accuracy * 100) / 100;
-    }
-
-    let error;
-    if (incorrectlyPredictedClass !== undefined) {
-      error = incorrectlyPredictedClass.count / classData.actual_class_doc_count;
-      error = Math.round(error * 100) / 100;
-    }
-
-    let col: any = {
+  confusionMatrixData.forEach(classData => {
+    const col: any = {
       actual_class: classData.actual_class,
       actual_class_doc_count: classData.actual_class_doc_count,
     };
+    const predictedClasses = classData.predicted_classes || [];
 
-    if (correctlyPredictedClass !== undefined) {
-      col = {
-        ...col,
-        predicted_class: correctlyPredictedClass.predicted_class,
-        [correctlyPredictedClass.predicted_class]: accuracy,
-        count: correctlyPredictedClass.count,
-        accuracy,
-      };
-    }
+    columns.push({ id: classData.actual_class });
 
-    if (incorrectlyPredictedClass !== undefined) {
-      col = {
-        ...col,
-        [incorrectlyPredictedClass.predicted_class]: error,
-        error_count: incorrectlyPredictedClass.count,
-      };
+    for (let i = 0; i < predictedClasses.length; i++) {
+      const predictedClass = predictedClasses[i].predicted_class;
+      const predictedClassCount = predictedClasses[i].count;
+      col[predictedClass] = predictedClassCount;
     }
 
     colData.push(col);
   });
 
-  const columns: any = [
+  return { columns, columnData: colData };
+}
+
+export function getTrailingControlColumns(numColumns: number, setShowFullColumns: any) {
+  return [
     {
-      id: ACTUAL_CLASS_ID,
-      display: <span />,
+      id: 'actions',
+      width: 60,
+      headerCellRender: () => `${numColumns} more`,
+      rowCellRender: function RowCellRender() {
+        const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
+        return (
+          <EuiPopover
+            isOpen={isPopoverOpen}
+            anchorPosition="upCenter"
+            button={
+              <EuiButtonIcon
+                aria-label="show actions"
+                iconType="boxesHorizontal"
+                color="text"
+                onClick={() => setIsPopoverOpen(!isPopoverOpen)}
+              />
+            }
+            closePopover={() => setIsPopoverOpen(false)}
+            ownFocus={true}
+          >
+            <EuiButtonEmpty onClick={() => setShowFullColumns(true)}>
+              <EuiText size="s" grow={false} textAlign="center">
+                Show all columns
```

Comment:
This message needs internationalizing.

Reply:
👍  added in 93a2deb48ce88f48679938a05835b67d1dab3e54